add validation of phonenumber

daschJulia opened the pull request:
It is needed to validate numbers if they are possible phone numbers.
So we do not only check if the phone number can be parsed but also
possible numbers for the country and region.

rjayasinghe commented:
Olle, thanks for the detailed review. I changed the spelling of userUUID and increased the branch coverage for the uniquenessvalidator. Having dedicated classes for phone number and mail will go into the next iteration as I will work on phone number formatting soon. ;)

daschJulia committed:
add validation of phonenumber

It is needed to validate numbers if they are possible phone numbers.
So we do not only check if the phone number can be parsed but also
possible numbers for the country and region.

rjayasinghe commented on src/main/java/net/contargo/types/telephony/validation/ValidPhoneNumberValidator.java:
```diff
@@ -49,6 +50,11 @@ public boolean isValid(String value, ConstraintValidatorContext cvc) {
                 isValid = false;
             }
+
+            if (phoneNumber.isValidNumber()) {
```
Shouldn't this be negated? This block now is entered for a valid number and than isValid is set false. Does not seem right to me.

daschJulia committed:
add correct validation, musst be negated

daschJulia committed:
remove tipo

rjayasinghe reviewed (approved):
Looks good to me. @contargo-development please merge
